pkg/asset/installconfig/basedomain.go

```diff
@@ -49,10 +51,14 @@ func (a *baseDomain) Generate(parents asset.Parents) error {
 			return err
 		}
 		a.BaseDomain = zone.Name
+		// XXX: It would be nice to check for forbidden/throttle error
+		// like aws and gcp here.
```

Comment:
we should open a github issue, or jira issue to track this future improvement and link that here. so as to better track it.

also azure sdk automatically retries throttle issues gracefully so we shouldn't need that here, forbidden would be nice.